### DISCUSSION TITLE
[Feature] Transfer a image from one user to another

### DISCUSSION THREAD

**Discussion by alberto8051** [immich-app/immich#1992] [Feature Request]
### The feature

Hi everyone, 

For me, one of the greatest features of immich is the multi-account. I'm trying to organize the family photos, so, sometimes, I need to delete a photo from an user and reimport again from another user. I think it would be useful to transfer the photo between users without deleting and reimport again. 

Thank you. 
Best Regards 
Alberto

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by avtechmjc** [comment]
I would find this useful too, as would anyone who needs to separate personal vs professional photos, or SFW and NSFW photos.

**Comment by ChilliGeologist** [comment]
I would also love this feature, at some stage some of my Wife's photos ended up in my photo folders and Immich has made it very easy to find these as my photo folder organisation isn't great. Having the option to tick a bunch of them and move them into her library instead of mine would be fantastic.

**Comment by aviv926** [comment]
I can think of a temporary solution that might give you an answer
In the user from which you want to transfer all the photos, mark all the photos you want to transfer and create a folder from them
Then with the option of sharing folders / or alternatively downloading and re-uploading
Transfer the data to the other user you want
When you are done with the main user, you can delete all the items from the album

It's a bit of a crooked way but it can be a good temporary solution

**Comment by xloss** [comment]
It's strange that this function hasn't been implemented in over a year. According to the description, it doesn't seem complicated to implement. And it's really very useful, so you don't have to re-upload photos.

I would also add here that a user with editing rights could also invite users, since when you change the owner, you lose control over the album.

**Reply by bo0tzz** [reply]
There's endless work to do, so it's a simple matter of priority. If it doesn't seem complicated to you, a PR would be welcome ;)

**Reply by joergbonsack** [reply]
@bo0tzz  I know that there is so much work and the community is so thankful for your work. 
I think this feature is really helpful. Are there any plans when it will be implemented?

**Reply by bo0tzz** [reply]
Nope

**Comment by fanuch** [comment]
Posting as this came up in my search.
I needed to transfer all photos from one user to another. This is not the exact nature of this feature request but it might help others.

Can do it with
```
docker exec -t immich_postgres psql -U postgres immich -c 'UPDATE person SET "ownerId" = "32f448f7-0f65-4293-986a-299ec4c58cb3" WHERE "ownerId" = "aea2260b-c821-45b2-aa5b-adb412cf3a23";'
docker exec -t immich_postgres psql -U postgres immich -c 'UPDATE album SET "ownerId" = "32f448f7-0f65-4293-986a-299ec4c58cb3" WHERE "ownerId" = "aea2260b-c821-45b2-aa5b-adb412cf3a23";'
docker exec -t immich_postgres psql -U postgres immich -c 'UPDATE assets SET "ownerId" = "32f448f7-0f65-4293-986a-299ec4c58cb3" WHERE "ownerId" = "aea2260b-c821-45b2-aa5b-adb412cf3a23";'
  ```
  
  Then 
  ```
  docker compose restart
  ```
  
  Where:
  - `aea2260b-c821-45b2-aa5b-adb412cf3a23` is the old ID
  - `32f448f7-0f65-4293-986a-299ec4c58cb3` is the new ID
  
  Note, I did not use tags or anything like that. These were the only tables that contained `ownerId` that had data for me
  
  ---
  
  A feature do to this through the frontend would be incredibly helpful

**Reply by mmomjian** [reply]
For anyone reading this, this is an old command that we think has led to database corruption and issues for people that have ran it in the past. The issue is not initially obvious but comes up with future upgrades/database changes.

We do not have any supported method to change ownership other than delete + reupload.

**Comment by deddc23efb** [comment]
--- I was confused by an admin directory that appears to contain photos.

**Comment by Ebnigma** [comment]
Would also love to see this feature. The Partner Sharing is great but i don't want to share all pictures in my library with my wife. I often take hobby photos that she does not care about so the imagined solution would be do create a "Family" user soley for the purpose of real family pictures and "move" these kind of photos to this account by "change owner". 
Then my wife and are use Family user by "Partner Sharing".
So moving images to "Partner Shared" accounts would be great.

**Comment by bo0tzz** [comment]
#1587
